explorer/interpreter/interpreter.cpp

```diff
@@ -182,7 +184,15 @@ class Interpreter {
       -> ErrorOr<Success>;
 
   auto CallDestructor(Nonnull<const DestructorDeclaration*> fun,
-                      Nonnull<const Value*> receiver) -> ErrorOr<Success>;
+                      Nonnull<const Value*> receiver,
+                      const LocationValue* class_addr) -> ErrorOr<Success>;
+
+  // If the given `method` has a self argument, bind it to `method_scope`.
```

Comment:
What do the other parameters mean? I'm particularly looking to understand the relationship between `receiver` and `receiver_addr`. It's especially puzzling to me that they seem to be identical except in the destructor case -- if we could arrange for them to always be identical, we could drop one of the parameters. That would be really nice, because 6 is kind of a lot.

Reply:
Added a comment with explanation.